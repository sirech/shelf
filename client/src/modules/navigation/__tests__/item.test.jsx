import React from 'react'
import renderer from 'react-test-renderer'

import Item from '../item'

const item = (activeYear = 2015) => {
  const props = {
    year: 2016,
    count: 2,
    activeYear: activeYear
  }

  return renderer.create(<Item {...props} />)
}

describe('components', () => {
  describe('Item', () => {
    it('renders', () => {
      expect(item().toJSON()).toMatchSnapshot()
    })

    it('renders for the active item', () => {
      expect(item(2016).toJSON()).toMatchSnapshot()
    })
  })
})
