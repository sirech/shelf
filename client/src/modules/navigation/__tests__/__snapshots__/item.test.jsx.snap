exports[`components Item renders 1`] = `
<a
  className="list-group-item list-group-item-action"
  onClick={[Function]}
  style={Object {}}>
  <span>
    2016
  </span>
  <span
    className="badge badge-warning badge-pill ml-auto">
    2
  </span>
</a>
`;

exports[`components Item renders for the active item 1`] = `
<a
  className="list-group-item list-group-item-action active"
  onClick={[Function]}
  style={Object {}}>
  <span>
    2016
  </span>
  <span
    className="badge badge-warning badge-pill ml-auto">
    2
  </span>
</a>
`;
